### PR TITLE
Speedup Shared Graphs

### PR DESCRIPTION
**Pull request by AkshayGoelTech**
Please refer https://github.com/Murali-group/GraphSpace/issues/331
This branch accomplishes speeding up the search graphs query to be incredibly fast
- We no longer need to retrieve information from postgres when getting shared graphs.
- Every graph in elasticsearch now has a list of users that this graph is shared with
- This list is very carefully updated whenever share access changes in any form

Note that this branch carefully handles the following 6 cases:
* User shares Graph with a group
* User unshares Graph with a group
* User added to a group
* User removed from a group
* Group is deleted
* Group is added

### CODE REVIEW

File: applications/users/controllers.py [hunk 1]
```diff
@@ -239,6 +243,17 @@ def get_group_members(request, group_id):
...
 	return members
 
+# This will retrieve the new list of users every graph in graph_ids is shared with
+# and then update elasticsearch.
+# graph_ids is None if a new member is added. ids are obtained from group_id
+# graphs_ids is not None if a group has been deleted. Calling function passes in a list of ids.
+def update_shared_users_elasticsearch(request, group_id, graph_ids=None):
+	if graph_ids is None:
+		graph_ids = [graph.graph_id for graph in graphs.controllers.get_graphs_by_group(request.db_session, group_id)]
+	for graph_id in graph_ids:
```
Comment (adbharadwaj) [inline applications/users/controllers.py]: Instead of updating graphs one at a time. Use batch update.

File: applications/users/controllers.py [hunk 2]
```diff
@@ -239,6 +243,17 @@ def get_group_members(request, group_id):
 	members = db.get_users_by_group(request.db_session, group_id)
 	return members
 
+# This will retrieve the new list of users every graph in graph_ids is shared with
+# and then update elasticsearch.
+# graph_ids is None if a new member is added. ids are obtained from group_id
+# graphs_ids is not None if a group has been deleted. Calling function passes in a list of ids.
+def update_shared_users_elasticsearch(request, group_id, graph_ids=None):
```
Comment (adbharadwaj) [inline applications/users/controllers.py]: Use triple quote commenting style everywhere

File: migrate_elasticsearch.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import json
+
+import sqlalchemy as sa
+from sqlalchemy import create_engine, ForeignKey
+from elasticsearch import Elasticsearch
+from graphspace.data_type import DataType
+
```
Comment (adbharadwaj) [inline migrate_elasticsearch.py]: Add comments on changes.

**Comment by jddanna** [comment]
Made new PR with this branch merged into it and referencing this PR. Thank you for your contributions!

**Pull request closed without merging**